src/pages/ImportData.js

```diff
@@ -0,0 +1,57 @@
+import { Fragment } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { pageColumn } from 'src/components/common'
+import { TopBar } from 'src/components/TopBar'
+import { DestinationProject } from 'src/pages/ImportTool'
+import { Orchestration } from 'src/libs/ajax'
+import * as Nav from 'src/libs/nav'
+import { Component } from 'src/libs/wrapped-components'
+
+
+class Importer extends Component {
+  render() {
+    // TODO(dmohs): This should be handled by the router.
+    const url = decodeURIComponent(this.props.url)
+    const { isImporting, selectedWorkspace } = this.state
+
+    return h(Fragment, [
+      h(TopBar, { title: 'Import Data' }),
+      // 23rem allows enough space for the opened selection box.
+      div({ style: { display: 'flex', minHeight: '23rem' } },
+        [
+          pageColumn('Importing', 5, div({}, [
+            div({ style: { overflowX: 'auto', whiteSpace: 'nowrap' } }, url)
+          ])),
+          pageColumn('Destination Project', 3,
+            h(DestinationProject, {
+              isImporting,
+              selectedWorkspace,
+              onWorkspaceSelected: selectedWorkspace => this.setState({ selectedWorkspace }),
+              import_: () => this.import_()
+            }))
+        ]
+      )
+    ])
+  }
+
+  async import_() {
+    this.setState({ isImporting: true })
+    const { selectedWorkspace: { value: { namespace, name } } } = this.state
+    const url = decodeURIComponent(this.props.url)
+
+    try {
+      await Orchestration.workspaces(namespace, name).importBagit(url)
+      Nav.goToPath('workspace', { namespace, name, activeTab: 'data' })
+    } finally {
+      this.setState({ isImporting: false })
+    }
+  }
+}
+
+
+export const addNavPaths = () => {
+  Nav.defPath('import-data', {
+    path: '/import-data?url=:url',
```

Comment:
`path-to-regexp' is usually used for the path portion only, and `query-string` is used alongside to handle query params. We could have it inject params as props, and it will automatically handle url decoding, etc.